packages/abi-coder/src/types/JsonAbi.ts

```diff
@@ -12,6 +12,12 @@ export interface JsonAbi {
 }
 
 export interface JsonAbiType {
+  /**
+   * concreteTypeId is a property from the new abi spec
+   * that we are adding to the old abi spec
+   * so that we can allow users to encode/decode concrete types on the `Interface`.
+   */
+  readonly concreteTypeId: string;
```

Comment:
How this will avoid breaking changes in the future?

If we return this type, then all the old attributes are also exposed:
https://github.com/FuelLabs/fuels-ts/blob/4893c569c65b41846ec836b32d14551e51f318da/packages/abi-coder/src/types/JsonAbi.ts#L21-L25

Reply:
This type is unavailable to users, but I removed it anyways in https://github.com/FuelLabs/fuels-ts/pull/2862/commits/5a1776f1ec4b612f648ee194f2564b86b518fb14.